### PR TITLE
popToTop

### PR DESCRIPTION
**Pull request by chirag04**
yet to test tho.

@skevy @brentvatne

### CODE REVIEW

File: src/ExNavigationReducer.js [hunk 1]
```diff
@@ -145,6 +145,27 @@ class ExNavigationReducer {
...
+    if (navigatorState.type === 'slidingTab') {
+      return _updateNavigator(
+        state,
+        navigatorUID,
+        {...navigatorState, index: 0 },
+      );
+    }
+
+    const routes = navigatorState.routes.slice(0, 1);
+    const newState = { ...navigatorState, index: 0, routes };
```
Comment (chirag04) [inline src/ExNavigationReducer.js]: probably rename to newNavigatorState.

**Comment by chirag04** [comment]
Ok tested. good to go.

Animation is not nice tho. `A, B, C, popToTop`, you can see it going from C to B to A as discrete steps instead of one linear pop.

could be a problem with upstream navigationExp.

**Comment by chirag04** [comment]
@brentvatne @skevy ping

**Comment by skevy** [comment]
@chirag04 can you confirm that after the popToTop reducer runs, the navigation state is correct?

**Comment by chirag04** [comment]
@skevy yes navigation state is correct after the reducer runs. not sure if i'm missing anything here.

if it's about the animation i figure it could be the dev mode.

**Comment by chirag04** [comment]
friendly ping :)

**Comment by brentvatne** [comment]
merged, thanks!

**Pull request closed without merging**